capsules/src/virtual_spi.rs

```diff
@@ -38,33 +38,46 @@ impl<'a, SPI: hil::spi::SpiMaster> MuxSPIMaster<'a, SPI> {
         if self.inflight.is_none() {
             let mnode = self.devices.iter().find(|node| node.operation.get() != Op::Idle);
             mnode.map(|node| {
-
-                match node.operation.get() {
+                self.spi.specify_chip_select(node.chip_select.get());
+                let op = node.operation.get();
+                // Need to set idle here in case callback changes state
+                node.operation.set(Op::Idle);
+                match op {
                     Op::Configure(cpol, cpal, rate) => {
 
                         // The `chip_select` type will be correct based on
                         // what implemented `SpiMaster`.
-                        self.spi.specify_chip_select(node.chip_select.get());
-
                         self.spi.set_clock(cpol);
                         self.spi.set_phase(cpal);
                         self.spi.set_rate(rate);
                     }
                     Op::ReadWriteBytes(len) => {
-
+                        // Only async operations want to block by setting
+                        // the devices as inflight.
```

Comment:
The inflight variable specifies whether there is an ongoing DMA transaction. While there is, you can't change the SPI configuration. Changing the configuration isn't asynchronous (you just update some control registers).